router/router.go

```diff
@@ -1288,9 +1328,16 @@ func (rt *HandleT) findWorker(job *jobsdb.JobT, throttledAtTime time.Time) (toSe
 		return nil
 	}
 
+	defer func() {
+		if toSendWorker == nil {
+			rt.throttler.Dec(parameters.DestinationID, userID, 1, throttledAtTime, throttler.ALL_LEVELS)
+		}
+	}()
```

Comment:
Added this since we need to decrement the throttler if no worker is found for the job.

As a general note, the current throttling logic (incrementing/decrementing) appears to be fragile.

cc @fracasula